_tasks/dnt.ts

```diff
@@ -148,16 +149,35 @@ await Promise.all([
     "target/npm/esm/main.js",
     (content) =>
       content
-        .replace(/^#!.+/, "#!/usr/bin/env -S node --loader ts-node/esm"),
+        .replace(/^#!.+/, "#!/usr/bin/env -S node --loader capi/ts-loader"),
   ),
   editFile(
     "target/npm/esm/_dnt.shims.js",
     (content) =>
       content
         .replace(/"@deno\/shim-deno"/g, `"./deps/shims/Deno.node.js"`),
   ),
+  editFile(
+    "target/npm/package.json",
+    (content) => {
+      const packageJson = JSON.parse(content)
+      packageJson.exports["./ts-loader"] = "./ts-loader.js"
+      return JSON.stringify(packageJson, undefined, 2)
+    },
+  ),
 ])
 
 async function editFile(path: string, modify: (content: string) => string) {
   await Deno.writeTextFile(path, modify(await Deno.readTextFile(path)))
 }
+
+async function writeTsLoader(path: string) {
+  const content = `
+import tsNode from "ts-node"
+
+const service = tsNode.create({ transpileOnly: true, compilerOptions: { module: "ESNext" } })
+
+export const { resolve, load, getFormat, transformSource } = tsNode.createEsmHooks(service)
```

Comment:
with this preconfigured `ts-loader`, the `tsconfig.json` is not needed to parse the `nets.ts`

Reply:
The `tsconfig` could specify many important things; I don't think we should hard-code this. I like the `capi/ts-loader` approach, but I think the file should just be
```suggestion
export * from "ts-node/esm"
```

Reply:
This is a slippery slope. The dev may have configured TS such that they expect to be able to use other language features in their `nets.ts` (ostensibly for loading in bins / instantiating netspecs to export). The user will be frustrated if their tsconfig is ignored / if the loader results in different behavior. On the flipside, we could load in their tsconfig and override `module`... but this could be problematic as well.

@tjjfvi said the following:

> The tsconfig could specify many important things; I don't think we should hard-code this.

I think I agree with this. We'll offer good debugging info in the case that they've misconfigured.

Reply:
> Also, replacing with
> 
> ```
> export * from "ts-node/esm"
> ```
> 
> is the same as executing `target/npm/esm/main.js` with
> 
> ```
> #!/usr/bin/env -S node --loader ts-node/esm
> ```

Except that it doesn't require downstream users to install ts-node themselves

Reply:
ok, I'll keep the custom loader with `export * from "ts-node/esm"`

and, I'll add a check in the `init` to require a `tsconfig.json` with `compilerOptions.module: ESNext`

keep in mind that the `tsconfig.json` will be required even if this is pure JS proyect.

@harrysolovay @tjjfvi Are you fine with the above?

Reply:
> `compilerOptions.module: ESNext`

We don't require this as of #1061 

> keep in mind that the `tsconfig.json` will be required even if this is pure JS project.

Ah, true, I forgot that we support `nets.js` files. Perhaps then, we do want a custom loader, which, if a `tsconfig.json` is present, uses `ts-node/esm`, and otherwise is a no-op.

Reply:
Let's defer making this loader better and keep it with `--loader ts-node/esm` for now

Reply:
for `--loader ts-node/esm` to work with the `main` Capi CLI entry and a `nets.ts` file, the following is required
- the `tsconfig.json` should exist
- `compilerOptions.module` needs to be `ES*` or `Node*`
- `compilerOptions.moduleResolution` needs to be `node*` or `bundler`

if that's ok, I'll add the checks to `capi init`

Reply:
I think that's out of scope for `capi init`, especially since we'll be improving the loader soon.

Reply:
the above is challenging if the `tsconfig.json` [extends](https://www.typescriptlang.org/tsconfig#extends) from another config

Reply:
Found another issue parsing the `tsconfig.json`.
`JSON.parse` does not work when there are comments.

Reply:
documented concerns in https://github.com/paritytech/capi/issues/1064